lua/entities/gmod_wire_dhdd.lua

```diff
@@ -14,11 +14,24 @@ function ENT:Initialize()
 	self.Outputs = WireLib.CreateOutputs( self, { "Memory [ARRAY]", "Size" } )
 	self.Inputs = WireLib.CreateInputs( self, { "Data [ARRAY]", "Clear", "AllowWrite" } )
 
-	self.Memory = {}
+	self.Size = 0
 	self.ROM = false
 	self.AllowWrite = true
 
 	self:SetOverlayText("DHDD")
+	self.MemoryMetatable = {
+		__len = function() return math.max(0, self.Size - 1) end,
```

Comment:
`__len` doesn't currently work on tables, though I guess it doesn't hurt to keep this if that's ever changed,